include/mav/Network.h

```diff
@@ -404,6 +411,18 @@ namespace mav {
             _heartbeat_message = std::nullopt;
         }
 
+        void setGetTimestampFunction(std::function<uint64_t(void)> function) {
+            _get_timestamp_function = function;
+        }
+
+        void setKey(std::array<uint8_t, 32> key) {
+            _key = key;
+        }
```

Comment:
Can you change this to a single function? A user would never just want to set a gettimestamp function or setting a key, what a user really wants is "enabling message signing". Could you make the API something like

```
network.enableMessageSigning("my-secret_key");
network.disableMessageSigning();
```

and 
```
network.enableMessageSigning("my-secret-key", [](){ /* my timestamp func */});
network.disableMessageSigning
```

Reply:
> A user would never just want to set a gettimestamp function or setting a key.

Why not? A user may wish to renegotiate a key multiple times, but the timestamp function probably wouldn't change.

Reply:
Changed as requested.